### PR TITLE
Add Unit Test For NegotiationResponse

### PR DESCRIPTION
**Pull request by zhangn49**
## Overview

Description:

Why should this be merged: 

Related issue(s) (if applicable): #<number>

## Checklist (Definition of Done):

- [x] There are no TODOs left in the code
- [x] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [x] Change is covered by automated tests
- [x] Public API has Javadoc

### CODE REVIEW

File: infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/receive/LogReplicationMetadataManager.java [hunk 1]
```diff
@@ -298,13 +298,13 @@ public void setSnapshotApplied(LogReplicationEntry entry) {
     }
 
     public String getMetadata() {
```
Comment (annym) [inline infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/receive/LogReplicationMetadataManager.java:300]: This is intended to be a friendly print of the metadata, right? can we move it to the override of the toString?

File: log-replication/src/main/java/org/corfudb/logreplication/runtime/CorfuLogReplicationRuntime.java [hunk 2]
```diff
@@ -77,6 +78,23 @@ public CorfuLogReplicationRuntime(@NonNull LogReplicationRuntimeParameters param
         corfuRuntime.connect();
     }
 
+    /**
+     * Constructor
+     *
+     * @param parameters runtime parameters
+     */
+    @VisibleForTesting
```
Comment (annym) [inline log-replication/src/main/java/org/corfudb/logreplication/runtime/CorfuLogReplicationRuntime.java:86]: Nan can you please rebase again on top of log-replication-feature-annym, this code changed a bit.

File: test/src/test/java/org/corfudb/logreplication/runtime/NegotiationResponseTest.java [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.RETURNS_DEEP_STUBS;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.BDDMockito.given;
+
+@SuppressWarnings("checkstyle:magicnumber")
```
Comment (annym) [inline test/src/test/java/org/corfudb/logreplication/runtime/NegotiationResponseTest.java:19]: Can you please add a general description to this suite. What we are "mocking", what is tested, unit? IT?

**Comment by xiaoqin2012** [comment]
I just realized that we need to check two cases:
1. while the receiver shows it has never done a transfer with all values -1 including the lastLogEntryTimestamp -1, and the sender has log tail -1, in this case, we can just start log entry sync and there is no snapshot sync required. We can do a snapshot sync, but the receiver metadata will not change as there is no data sent over , only snapshot-start and snapshot-end will be sent over.
2. From scenario 1, we can have case that all snapshot related metadata values are -1, but lastLogEntryTimestamp is 100, in this case, if logEntry 101 is not trimmed at the sender, we should start logEntrySync, there is no full snapshot sync is required.

**Pull request closed without merging**